pandas/io/formats/style_render.py

```diff
@@ -2379,16 +2395,55 @@ def _escape_latex_math(s):
     str :
         Escaped string
     """
-    s = s.replace(r"\$", r"rt8§=§7wz")
-    pattern = re.compile(r"\$.*?\$")
-    pos = 0
-    ps = pattern.search(s, pos)
-    res = []
-    while ps:
-        res.append(_escape_latex(s[pos : ps.span()[0]]))
-        res.append(ps.group())
-        pos = ps.span()[1]
+
+    def _math_mode_with_dollar(s):
```

Comment:
Could probably move the methods outside of the `_escape_latex_math` to avoid redefining them each function call.

Reply:
Good point, I moved the methods outside of the _escape_latex_math. 

I changed a little bit the logic of the `_escape_latex_math`. Now if we have a combination of dollar signs and brackets, the dollar sign will be applied.